plugins/languages/otter.nix

```
@@ -83,4 +83,14 @@ helpers.neovim-plugin.mkNeovimPlugin config {
       When true, otter handles these cases fully. This is a (minor) performance hit.
     '';
   };
+
+  extraConfig = cfg: {
+    warnings =
+      lib.optional (cfg.enable -> config.plugins.treesitter.settings.highlight.enable == null)
```

```suggestion
      lib.optional (cfg.enable && config.plugins.treesitter.settings.highlight.enable == null)
```

Using `->` here would show the warning whenever `cfg.enable == false`

---

That's not what i'm observing in the local tests. If I set otter to disabled in the tests  I dont get the warning.

---

That's probably because `highlight.enable == null` when you're testing?

`a -> b` is equivalent to `!a || b`. Here I'm pretty sure we want to show the warning only when _both_ `enable` is true and `highlight.enable` is null.

---

I added ``plugins.treesitter.settings.highlight.enable = true; to the default test and then set otter enable = false.
``

---

I tried a few different combinations in the otter default test. They all didn't result in a warning. Is the mkNeovimPlugin doing something to prevent it?

- `plugins.treesitter.settings.highlight.enable = true;` and `otter.enable = false`
- `plugins.treesitter.settings.highlight.enable = false;` and `otter.enable = false`
- `plugins.treesitter.settings.highlight.enable = null;` and `otter.enable = false`

---

That said, I see the same behavior doing it with -> and && so I can change it, just wanted to point out I wasn't seeing the behavior mentioned.